### DISCUSSION TITLE
Path finding based solver/simplifier

### DISCUSSION THREAD

**Discussion by mcpiroman** [sympy/sympy#24486] [Ideas]
I remember reading some comment in the sympy's source suggesting an alternative implementation of solver (or was it just simplifier) that would repeatedly apply a set of transformation rules and find a shortest path to the result based on the 'simpleness' heuristic.

I really like the idea but I cannot find that comment nor relevant issue anymore and neither I see any advance in this field in the source. Was / is such an solution being considered?

This is most probably akin to the #12233 (and the rule based integration in general), possibly except that here it is less clear which rules to select next and even to decide when to finish execution, as there often will be even-more applicable rules.

**Comment by smichr** [comment]
The Fu methodology implements this idea in simplifying trigonometric expressions.

**Reply by smichr** [reply]
There was also this: https://groups.google.com/g/sympy/c/57IwNoIhpyo/m/ezbshCT-AQAJ
